src/porepy/models/verification_setups/terzaghi.py

```diff
@@ -0,0 +1,819 @@
+        ax.legend(fontsize=14)
+        ax.grid()
+        plt.show()
+
+
+class ModifiedGeometry(pp.ModelGeometry):
+    """Define geometry of the verification setup."""
+
+    params: dict
+    """Simulation model parameters."""
+
+    def set_md_grid(self) -> None:
+        """Create the mixed-dimensional grid base on two-dimensional Cartesian grid."""
+        height = self.params.get("height", 1.0)  # [m]
+        num_cells = self.params.get("num_cells", 20)
+        ls = 1 / self.units.m
+        phys_dims = np.array([height, height]) * ls
+        n_cells = np.array([1, num_cells])
+        self.domain_bounds = pp.geometry.bounding_box.from_points(
+            np.array([[0, 0], phys_dims]).T
+        )
+        sd: pp.Grid = pp.CartGrid(n_cells, phys_dims)
+        sd.compute_geometry()
+        self.mdg = pp.meshing.subdomains_to_mdg([[sd]])
+
+
+# ----> Boundary conditions
+class ModifiedBoundaryConditionsMechanicsTimeDependent(
+    poromechanics.BoundaryConditionsMechanicsTimeDependent,
+):
+    mdg: pp.MixedDimensionalGrid
+    """Mixed-dimensional grid"""
+
+    domain_boundary_sides: Callable[[pp.Grid], pp.bounding_box.DomainSides]
+    """Named tuple containing the boundary sides indices."""
+
+    stress_keyword: str
+    """Keyword for the mechanical subproblem."""
+
+    bc_values_mechanics_key: str
+    """Keyword for accessing the boundary values for the mechanical subproblem."""
+
+    params: dict
+    """Parameter dictionary of the verification setup."""
+
+    def bc_type_mechanics(self, sd: pp.Grid) -> pp.BoundaryConditionVectorial:
+        """Define type of boundary conditions.
+
+        Parameters:
+            sd: Subdomain grid.
+
+        Returns:
+            bc: Boundary condition representation. Neumann on the north, Dirichlet on
+              south, and rollers on the sides.
+
+        """
+        # Inherit bc from parent class. This sets all bc faces as Dirichlet.
+        bc = super().bc_type_mechanics(sd=sd)
+
+        # Get boundary sides, retrieve data dict, and bc object
+        _, east, west, north, *_ = self.domain_boundary_sides(sd)
+
+        # East side: Roller
+        bc.is_neu[1, east] = True
+        bc.is_dir[1, east] = False
+
+        # West side: Roller
+        bc.is_neu[1, west] = True
+        bc.is_dir[1, west] = False
+
+        # North side: Neumann
+        bc.is_neu[:, north] = True
+        bc.is_dir[:, north] = False
+
+        # South side: Dirichlet (already set thanks to inheritance)
+
+        return bc
+
+    def time_dependent_bc_values_mechanics(
+        self, subdomains: list[pp.Grid]
+    ) -> np.ndarray:
+        """Boundary values for mechanics.
+
+        Parameters:
+            subdomains: List of subdomains on which to define boundary conditions.
+
+        Returns:
+            Array of boundary values. Only non-zero values are the ones associated to
+              the North side of the domain.
```

Comment:
There seems to be an indentation mistake here and some other places.

Reply:
Fixed.